src/mvba/abba/tests.rs

```diff
@@ -558,3 +558,147 @@ fn test_normal_case_two_rounds() {
+            )));
+            let vcbc = Abba::new(
+                tag.clone(),
+                node_id,
+                public_key_set.clone(),
+                key_share,
+                broadcaster,
+            );
+            (node_id, vcbc)
+        }));
+
+        Net {
+            secret_key_set,
+            tag,
+            nodes,
+            queue: Default::default(),
+        }
+    }
+
+    fn node_mut(&mut self, id: NodeId) -> &mut Abba {
+        self.nodes.get_mut(&id).unwrap()
+    }
+
+    fn enqueue_bundles_from(&mut self, id: NodeId) {
+        let (send_bundles, bcast_bundles) = {
+            let mut broadcaster = self.node_mut(id).broadcaster.borrow_mut();
+            let send_bundles = broadcaster.take_send_bundles();
+            let bcast_bundles = broadcaster.take_broadcast_bundles();
+            (send_bundles, bcast_bundles)
+        };
+
+        for (recipient, bundle) in send_bundles {
+            let bundle: Bundle =
+                bincode::deserialize(&bundle).expect("Failed to deserialize bundle");
+            self.queue.entry(recipient).or_default().push(bundle);
+        }
+
+        for bundle in bcast_bundles {
+            for recipient in self.nodes.keys() {
+                let bundle: Bundle =
+                    bincode::deserialize(&bundle).expect("Failed to deserialize bundle");
+                self.queue.entry(*recipient).or_default().push(bundle);
+            }
+        }
+    }
+
+    fn drain_queue(&mut self) {
+        while !self.queue.is_empty() {
+            for (recipient, queue) in std::mem::take(&mut self.queue) {
+                let recipient_node = self.node_mut(recipient);
+
+                for bundle in queue {
+                    let msg: Message = bincode::deserialize(&bundle.message)
+                        .expect("Failed to deserialize message");
+
+                    println!("Handling message: {msg:?}");
+                    recipient_node
+                        .receive_message(bundle.sender, msg)
+                        .expect("Failed to receive msg");
+                }
+
+                self.enqueue_bundles_from(recipient);
+            }
+        }
+    }
+
+    #[allow(unused)]
+    fn deliver(&mut self, recipient: NodeId, index: usize) {
+        if let Some(msgs) = self.queue.get_mut(&recipient) {
+            if msgs.is_empty() {
+                return;
+            }
+            let index = index % msgs.len();
+
+            let bundle = msgs.swap_remove(index);
+            let msg: Message =
+                bincode::deserialize(&bundle.message).expect("Failed to deserialize message");
+
+            let recipient_node = self.node_mut(recipient);
+            recipient_node
+                .receive_message(bundle.sender, msg)
+                .expect("Failed to receive message");
+            self.enqueue_bundles_from(recipient);
+        }
+    }
+}
+
+#[test]
+fn test_net_happy_path() {
+    let proposer = 1;
```

Comment:
Any way to work on byzantine votes here?

Reply:
This is the happy path test, showing the behaviour when everyone is voting the same way honestly.